superset/jinja_context.py

```diff
@@ -229,22 +277,28 @@ def __init__(
             "filter_values": filter_values,
             "form_data": {},
         }
+
         self._context.update(kwargs)
         self._context.update(jinja_base_context)
+
         if self.engine:
             self._context[self.engine] = self
-        self._env = SandboxedEnvironment()
 
-    def process_template(self, sql: str, **kwargs: Any) -> str:
-        """Processes a sql template
 
-        >>> sql = "SELECT '{{ datetime(2017, 1, 1).isoformat() }}'"
-        >>> process_template(sql)
-        "SELECT '2017-01-01T00:00:00'"
-        """
-        template = self._env.from_string(sql)
-        kwargs.update(self._context)
-        return template.render(kwargs)
+class SafeJinjaTemplateProcessor(BaseTemplateProcessor):
+    def set_context(self, **kwargs: Any) -> None:
+        extra_cache = ExtraCache(self._extra_cache_keys)
+        self._context = {
+            "url_param": partial(safe_proxy, extra_cache.url_param),
+            "current_user_id": partial(safe_proxy, extra_cache.current_user_id),
+            "current_username": partial(safe_proxy, extra_cache.current_username),
+            "cache_key_wrapper": partial(safe_proxy, extra_cache.cache_key_wrapper),
+            "filter_values": partial(safe_proxy, filter_values),
+            "form_data": {},
```

Comment:
Wondering why we need this [empty dict] key (?)

Reply:
Not sure, likely legacy support? I'll remove.